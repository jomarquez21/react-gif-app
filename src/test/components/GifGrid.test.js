import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from './../../components/GifGrid';
// import toJson from 'enzyme-to-json';
import { useFetchGifs } from './../../hooks/useFetchGifs';
jest.mock('./../../hooks/useFetchGifs');

describe('Test funtional on `GifGrid`', () => {
  const category = 'Jiraya'
  
  test('Deberia renderizar correctamente', () => {
    // simulamos la respuesta por defecto.
    useFetchGifs.mockReturnValue({data: [], loading: true});
    const component = shallow(<GifGrid category={category} />);
    expect(component).toMatchSnapshot();
  });
  
  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [{
      id: 'abc',
      url: 'https://google.com',
      title: 'Any message',
    }];
    // simulamos la respuesta con los datos buscado en la api.
    useFetchGifs.mockReturnValue({data: gifs, loading: false});

    const component = shallow(<GifGrid category={category} />);

    expect(component).toMatchSnapshot();
    expect(component.find('p').exists()).toBe(false);
    expect(component.find('GifGridItem').length).toBe(gifs.length);
  });
});
