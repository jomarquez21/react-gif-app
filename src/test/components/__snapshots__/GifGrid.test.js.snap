// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test funtional on \`GifGrid\` Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Jiraya
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Any message"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Test funtional on \`GifGrid\` Deberia renderizar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Jiraya
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
